job.py

```diff
@@ -17,73 +19,64 @@ def __init__(self, url, email, password, catalogueDB, sourceDB):
-        newSubcohorts = self.__download('Subcohorts')
-        newPartners = self.__download('Partners')
-        newPublications = self.__download('Publications')
-
-        # # 4) Add/Upload to catalog
-        self.__uploadIfSet('Publications', newPublications)
-        self.__uploadIfSet('Cohorts', newCohorts)
-        self.__uploadIfSet('Documentation', newDocumentation)
-        self.__uploadIfSet('Contacts', newContacts)
-        self.__uploadIfSet('Contributions', newContributions)
-        self.__uploadIfSet('Subcohorts', newSubcohorts)
-        self.__uploadIfSet('CollectionEvents', newCollectionEvents)
-        self.__uploadIfSet('Partners', newPartners)
-
-    def __uploadIfSet(self, table, data):
+    def uploadIfSet(self, table, data):
         """ Upload staging data to catalogue if staging table contains data (else skip) """
         if data is None:
             return
         uploadResp = self.catalogue.uploadCSV(table, data)
-        log.info('upload ' + table +' ; ' + str(uploadResp))
+        log.info('upload ' + table + ' ; ' + str(uploadResp))
 
-    def __download(self, table):
+    def download(self, table):
         """ Download staging data or return None in case of zero rows """
         countResp = self.staging.query('query Count{' + table + '_agg { count }}')
         if countResp[table + '_agg']['count'] > 0:
-          return self.staging.downLoadCSV(table)
+            return self.staging.downLoadCSV(table)
         else:
-          return None
-    
-    def __fetchCohortPid(self, staging, schemaName):
-        """ Fetch first cohort and return pid or else fail """
-        cohortsResp = staging.query(Path('cohorts.gql').read_text())
-        if "Cohorts" in cohortsResp:
-            if len(cohortsResp['Cohorts']) != 1:
-                log.error('Expected a single cohort in staging area "' + schemaName + '" but found ' + str(len(cohortsResp['Cohorts'])))
+            return None
+
+    def fetchCohortPid(self, staging, schemaName):
+        """ Fetch first cohort and return pid or else fail
+        Not all staging areas (network staging areas, SharedStaging) contain a table 'Cohorts',
+        therefore a try/except is used here.
+        """
+        try:
+            cohortsResp = staging.query(Path('./graphql-queries/' + 'Cohorts.gql').read_text())
+            if "Cohorts" in cohortsResp:
+                if len(cohortsResp['Cohorts']) != 1:
+                    log.warning('Expected a single cohort in staging area "' + schemaName + '" but found ' + str(
+                        len(cohortsResp['Cohorts'])))
+                    return None
+            else:
+                log.warning('Expected a single cohort in staging area "' + schemaName + '" but found none')
                 return None
-        else:
-            log.error('Expected a single cohort in staging area "' + schemaName + '" but found none')
+
+            return cohortsResp['Cohorts'][0]['pid']
+        except KeyError:
```

Comment:
do not use try catch for controle flow , suggest to query the model is needed , and use query info to make decision of fetching stuff

Reply:
in python try/except is much used for control flow. query fails for certain sources if there is no 'Cohorts' table or 'Models' table, this is a handy way to catch that.